### PR TITLE
Feature/musl update1.1.20

### PR DESCRIPTION
**Pull request by berinpaul**
# musl library update

- musl library is updated to the  version 1.1.20

### Details of the newly added tests are given below :

sl no. |Test_name | compile | run | comment |
:---:|:---|:--:|:--:|:----|
1 | 3rdparty/musl/libc-test/src/functional/strftime.c | yes | no | Test uses the strftime function which is currently unsupported |
2 | 3rdparty/musl/libc-test/src/functional/tls_local_exec.c | yes | no | Thread local storage unsupported |
3 | 3rdparty/musl/libc-test/src/regression/pthread_atfork-errno-clobber.c | no | no | Refers to unsupported symbols in pthread.h |
4 | 3rdparty/musl/libc-test/src/regression/setvbuf-unget.c | yes | yes | - |

### CODE REVIEW

**Comment by berinpaul** [comment]
Update requires minor additions in the mbed test framework. Changes can be found [here](https://github.com/Microsoft/openenclave/pull/931/commits/b8f8beeeee6de1118df92fd0644f0aab581c0f96)

File: tests/libc/tests.unsupported [hunk 1]
```diff
@@ -71,22 +77,20 @@
...
+../../3rdparty/musl/libc-test/src/regression/regex-backref-0.c
+../../3rdparty/musl/libc-test/src/regression/regex-bracket-icase.c
+../../3rdparty/musl/libc-test/src/regression/regexec-nosub.c
+../../3rdparty/musl/libc-test/src/regression/regex-ere-backref.c
 ../../3rdparty/musl/libc-test/src/regression/regex-escaped-high-byte.c
+../../3rdparty/musl/libc-test/src/regression/regex-negated-range.c
 ../../3rdparty/musl/libc-test/src/regression/rewind-clear-error.c
 ../../3rdparty/musl/libc-test/src/regression/rlimit-open-files.c
 ../../3rdparty/musl/libc-test/src/regression/setenv-oom.c
+../../3rdparty/musl/libc-test/src/regression/setvbuf-unget.c
```
Comment (CodeMonkeyLeet) [inline tests/libc/tests.unsupported]: This test is in the supported list and should be removed from the unsupported list.

**Comment by berinpaul** [comment]
bors try

**Pull request merged**